WIP: using IcebergSink for dynamic tables

arkadius opened the pull request.

rodmeneses commented on gradle.properties:
```diff
@@ -16,7 +16,7 @@
 jmhJsonOutputPath=build/reports/jmh/results.json
 jmhIncludeRegex=.*
-systemProp.defaultFlinkVersions=1.20
+systemProp.defaultFlinkVersions=1.19
```
please remember to bring this back to 1.20

amogh-jahagirdar committed:
API, AWS: Retry S3InputStream reads (#10433)

Co-authored-by: Jack Ye <yzhaoqin@amazon.com>
Co-authored-by: Xiaoxuan Li <xioxuan@amazon.com>

laurentgo committed:
AWS: Fix AWS doc URL (#11198)

Fix AWS doc URL in S3FileIOProperties to point to the international
version instead of the Chinese version.

dependabot[bot] committed:
Build: Bump mkdocs-macros-plugin from 1.0.5 to 1.2.0 (#11189)

Bumps [mkdocs-macros-plugin](https://github.com/fralau/mkdocs_macros_plugin) from 1.0.5 to 1.2.0.
- [Release notes](https://github.com/fralau/mkdocs_macros_plugin/releases)
- [Changelog](https://github.com/fralau/mkdocs-macros-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/fralau/mkdocs_macros_plugin/compare/v1.0.5...v1.2.0)

---
updated-dependencies:
- dependency-name: mkdocs-macros-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

aokolnychyi committed:
Core: Add rewritten delete files to write results (#11203)

aokolnychyi committed:
Core: Support iterating over positions in PositionDeleteIndex (#11202)

aleenamg21-1 committed:
Spark: Added merge schema as spark configuration (#9640)

aokolnychyi committed:
Core: Support merging in PositionDeleteIndex (#11208)

wypoon committed:
Core: Remove unused code for streaming position deletes (#11175)

jbonofre committed:
Build: Upgrade to Gradle 8.10.2 (#11212)

ajantha-bhat committed:
Core: Add a util to compute partition stats (#11146)

aihuaxu committed:
Parquet: update PruneColumns to inherit from TypeWithSchemaVisitor to have Iceberg type (#11179)

amogh-jahagirdar committed:
Core: Replace use of CharSequenceMap in DeleteFileIndex with String (#11199)

anuragmantri committed:
[Core] Fix TestFastAppend.testAddManyFiles() (#11218)

ajantha-bhat committed:
Spark: Deprecate SparkAppenderFactory (#11076)

Commit:
Tests for the Flink Sink v2-based implemenation (IcebergSink)
